545 bug i18n is a server side singleton

myieye opened the pull request:
Resolves #545

hahn-kev commented on frontend/src/routes/(unauthenticated)/sandbox/i18n/ssr-only/+page.ts:
```diff
@@ -0,0 +1,10 @@
+import type { PageLoadEvent } from './$types';
+
+export const csr = false;
```
minor thing, but on this page the loading indicator never goes away, probably because of this.

myieye replied:
Yeah. I can't think of an easy way to fix that. 🤷

hahn-kev reviewed (changes requested):
looks good, I left a couple comments. I didn't do a super through test but I did look at it. Let me know if you want me to check more.

myieye commented on frontend/src/lib/i18n/index.ts:
```diff
@@ -40,9 +87,11 @@ export function tScoped<Scope extends I18nScope>(scope: Scope): Readable<(key: D
 }
 
+export type Translater = (key: I18nKey, values?: InterpolationValues) => string;
+type I18n = Readable<Translater>;
```
Wow. Good question. It seems that `typeof I18n` uses the imported one...black magic.

myieye commented:
So, I sort of overlooked the fact that I added type safety to all of our translations in my previous changes. As a result, I actually found a bug. So, instead of chaning the type-safety to a loose-autocomplete (which I almost did), I decided that I like the type safety, so I extended our translation API to allow dynamic sub-keys while still type-checking the root of a key.

I also addressed all your great comments, except I'm continuing to ignore the never dissappearing loading idicator on CSR-only pages.

hahn-kev reviewed (approved):
Nice work!
